### DISCUSSION TITLE
dbt-core/redshift library update and overflow of logs

### DISCUSSION THREAD

**Discussion by ssillaots-boku** [dagster-io/dagster#22460] [Q&A]
After updating `dbt-core` (to 1.8.2) and `dbt-redshift` (to 1.8.0) I get incredible amount of logs. Thousands of rows compared to couple rows preupdate. 

Different setups I've tried:
* Dagster is 1.7.7 and
  * `dbt-redshift` 1.7.7 and `dbt-core` 1.7.16 ✅ (just a couple of warnings)
  * `dbt-redshift` 1.8.0 and `dbt-core` 1.8.2 :x: 
* Python is 3.10.13

This clearly showcases that aforementioned libraries are the culprit. But the warnings seem to be related to Dagster code.
An example of the logs I get (here are only a few but locally I get thousands):
```
.venv/lib/python3.10/site-packages/dagster/_core/definitions/auto_materialize_policy.py:213: ExperimentalWarning: Class `AutoMaterializePolicy` is experimental. It may break in future versions, even between dot releases. To mute warnings for experimental functionality, invoke warnings.filterwarnings("ignore", category=dagster.ExperimentalWarning) or use one of the other methods described at https://docs.python.org/3/library/warnings.html#describing-warning-filters.
.venv/lib/python3.10/site-packages/dagster/_core/definitions/backfill_policy.py:61: ExperimentalWarning: Class `BackfillPolicy` is experimental. It may break in future versions, even between dot releases. To mute warnings for experimental functionality, invoke warnings.filterwarnings("ignore", category=dagster.ExperimentalWarning) or use one of the other methods described at https://docs.python.org/3/library/warnings.html#describing-warning-filters.
.venv/lib/python3.10/site-packages/dagster/_core/definitions/auto_materialize_rule.py:92: DeprecationWarning: Class `MaterializeOnRequiredForFreshnessRule` is deprecated and will be removed in 1.8. Lazy auto-materialize is deprecated, in favor of explicit cron-based scheduling rules. Additional alternatives to replicate more of the lazy auto-materialize behavior will be provided before this is fully removed.
```
A puzzling situation. I've also tried suppressing the warnings with 👇 in the code location's init file but still I get loads of warnings.
```
warnings.filterwarnings("ignore", category=DeprecationWarning, append=True)
warnings.filterwarnings("ignore", category=ExperimentalWarning, append=True)
```
Any recommendations how to solve it?

**Comment by garethbrickman** [comment]
Could you try this syntax:

```python
import warnings
from dagster import ExperimentalWarning

warnings.filterwarnings("ignore", category=DeprecationWarning)
warnings.filterwarnings("ignore", category=ExperimentalWarning)
```

You may need to add these in other files,[ as this user mentions here](https://github.com/dagster-io/dagster/discussions/17520#discussioncomment-7557252)

**Reply by ssillaots-boku** [reply]
Thanks for the suggestion. 
I'll go crazy if I start adding warnings block to specific files.
I've tried to add it to a single file per code location but I get weird behaviour depending on the code location. For some code locations adding it to a singular file is enough but for others it isn't.
Seems like there isn't a way to get rid of the messages easily. Hopefully next versions will fix it somehow.

**Reply by garethbrickman** [reply]
I think this PR will be the fix: https://github.com/dagster-io/dagster/pull/22528
